File: src/01-easy/lumen/solution.cs
using System;
class Solution{static void Main(){
    int n=int.Parse(Console.ReadLine()),l=int.Parse(Console.ReadLine());
    var g=new string[n,n];for(int i=0;i<n;i++){var r=Console.ReadLine().Split(' ');for(int j=0;j<n;j++)g[i,j]=r[j];}
    int d=0;
    for(int r=0;r<n;r++)for(int c=0;c<n;c++){bool lit=false;
        for(int r2=0;r2<n&&!lit;r2++)for(int c2=0;c2<n&&!lit;c2++)
            if(g[r2,c2]=="C"&&Math.Max(Math.Abs(r-r2),Math.Abs(c-c2))<l)lit=true;
        if(!lit)d++;}
    Console.WriteLine(d);
}}

File: src/01-easy/lumen/solution.vb
Module Solution
    Sub Main()
        Dim n=CInt(Console.ReadLine()),l=CInt(Console.ReadLine())
        Dim g(n-1)() As String
        For i=0 To n-1:g(i)=Console.ReadLine().Split(" "c):Next
        Dim d=0
        For r=0 To n-1:For c=0 To n-1
            Dim lit=False
            For r2=0 To n-1:For c2=0 To n-1
                If g(r2)(c2)="C" AndAlso Math.Max(Math.Abs(r-r2),Math.Abs(c-c2))<l Then lit=True
            Next:Next
            If Not lit Then d+=1
        Next:Next
        Console.WriteLine(d)
    End Sub
End Module
